wof/wofpy_config.py

```diff
@@ -16,62 +16,82 @@
 
 Usage:
     wofpy_config INDIR
-    wofpy_config INDIR [--mode=<development,production>]
+    wofpy_config INDIR [--mode=<development,production>] [--overwrite=<soft,hard,none>]
 
-    wofpy_config (-h | --help | -v | --version)
+    wofpy_config (-h | --help | -v | --version | -m | --mode | -o | --overwrite)
 
 Examples:
     wofpy_config wofpyserver
+    wofpy_config wofpyserver --mode=production --overwrite=soft
 
 Arguments:
-  directory     Configuration directory.
+  directory               Configuration directory.
 
 Options:
-  -h --help     Show this screen.
-  -v --version  Show version.
-  --mode=development   deploy mode [default: development]
+  -h --help               Show this screen.
+  -v --version            Show version.
+  -m --mode=development   Deploympent mode [default: development].
+  -o --overwrite=none     Overwrite everything *hard*,
+                          or only adds extra files to an exsting directory *soft*
+                          [default: none].
 """
 
 _ROOT = os.path.abspath(os.path.dirname(wof.__file__))
 _CONFIG = os.path.join(_ROOT, 'examples', 'production_configs')
 _ODM2_TIMESERIES = os.path.join(_ROOT, 'examples', 'flask', 'odm2', 'timeseries')
 
 
-def makedirs(directory):
+def makedirs(directory, overwrite=None):
     if not os.path.exists(directory):
         os.makedirs(directory)
+    elif os.path.exists(directory) and overwrite in ['soft', 'hard']:
+        if overwrite == 'hard':
+            shutil.rmtree(directory)
+            os.makedirs(directory)
     else:
         msg = 'Directory {} already exists.'.format
         raise ValueError(msg(directory))
 
 
-def copytree(src, dst, symlinks=False, ignore=None):
+def copytree(src, dst, symlinks=False, ignore=None, overwrite='none'):
     for item in os.listdir(src):
         if '.pyc' in item:
             continue
         s = os.path.join(src, item)
         d = os.path.join(dst, item)
+        if os.path.exists(d) and overwrite == 'soft':
+            continue
```

Comment:
Here we leave anu existing file untouch. This should not modify the user existing configuration.